api/v1/invitations/PKPInvitationController.php

```diff
@@ -0,0 +1,191 @@
+        
+        Route::put('decline/{invitationId}/key/{key}', $this->declineInvitation(...))
+            ->name('invitation.declineInvitation')
+            ->whereNumber('invitationId');
+    }
+
+    /**
+     * @copydoc \PKP\core\PKPBaseController::authorize()
+     */
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        return true;
+    }
+
+    /**
+     * Get a single invitation
+     */
+    public function get(Request $illuminateRequest): JsonResponse
+    {
+        $invitationKey = $illuminateRequest->route('key');
+        $invitationId = (int) $illuminateRequest->route('invitationId');
+
+        $invitation = Repo::invitation()
+            ->getByIdAndKey($invitationId, $invitationKey);
+    
+        if (is_null($invitation)) {
+            return response()->json([
+                'error' => __('api.invitations.404.invitationNotFound')
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        return response()->json($invitation, Response::HTTP_OK);
+    }
+
+    /**
+     * Update an invitation payload
+     */
+    public function updatePayload(Request $illuminateRequest): JsonResponse
+    {
+        $invitationKey = $illuminateRequest->route('key');
+        $invitationId = (int) $illuminateRequest->route('invitationId');
+
+        $invitation = Repo::invitation()
+            ->getByIdAndKey($invitationId, $invitationKey);
+    
+        if (is_null($invitation)) {
+            return response()->json([
+                'error' => __('api.invitations.404.invitationNotFound')
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        // Attempt to update the invitation payload in the database
+        try {
+            $updatedInvitation = Repo::invitation()->updatePayload($invitation, $illuminateRequest->input());
+
+            return response()->json([
+                'message' => __('api.invitations.200.invitationUpdatedSuccessfully'),
+                'invitation' => $updatedInvitation
+            ], Response::HTTP_OK);
+        } catch (\Exception $e) {
+            return response()->json([
+                'error' => __('api.invitations.500.invitationFailedToUpdate')
+            ], Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    /**
+     * Accept an invitation
+     */
+    public function acceptInvitation(Request $illuminateRequest): JsonResponse
+    {
+        $invitationKey = $illuminateRequest->route('key');
+        $invitationId = (int) $illuminateRequest->route('invitationId');
+
+        $invitation = Repo::invitation()
+            ->getByIdAndKey($invitationId, $invitationKey);
+    
+        if (is_null($invitation)) {
+            return response()->json([
+                'error' => __('api.invitations.404.invitationNotFound')
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        // Attempt to acceptHandle
+        try {
+            $invitation->acceptHandle();
```

Comment:
Are these accept/decline functions intended to replace the similar calls on the page handler?